### PR TITLE
Hide AMP Noloading toggle if block does not have specified setting

### LINKED ISSUE #4555: Deprecate AMP Noloading setting
## Feature description

The “AMP Noloading” setting is not user-friendly:

image

For one, users will most likely not understand what this is doing, unless they are AMP developers. What's more is that this setting is probably never used. It wouldn't be serving the needs of the 80%. Therefore, I suggest that we remove the toggle entirely, unless (perhaps) the block currently has the setting specified: this will allow existing blocks to manipulate the layout, but for newly-added blocks the setting would go away.

Aside: Toggle currently appears for the YouTube embed block but it has no effect.

We also want to minimize “AMP” anywhere in the UI. It's not something that authors should need care about at all. If we do keep the setting, it should be moved to the Advanced panel alongside the CSS classes.

If a site needs this setting, it could be added via a plugin.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by pierlon**
## Summary

Fixes #4555

## Checklist

- [x] My pull request is addressing an open issue (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: assets/src/block-editor/helpers/index.js [hunk 1]
```diff
@@ -413,13 +413,19 @@ AmpLayoutControl.propTypes = {
...
+ * @deprecated As of v2.1. Blocks with the `ampNoLoading` attribute will still be able to use the control.
+ *
  * @param {Object} props Props.
  *
  * @return {ReactElement} Element.
  */
 const AmpNoloadingToggle = ( props ) => {
 	const { attributes: { ampNoLoading }, setAttributes } = props;
 
+	if ( ! ampNoLoading ) {
```
Comment (johnwatkins0) [inline assets/src/block-editor/helpers/index.js]: Should this be:

```suggestion
	if ( undefined === ampNoLoading ) {
```

If it's just `! ampNoLoading`, then the toggle will be hidden on existing blocks that have it set to false. And if an existing block has it set to true and it is toggled to false, the toggle will disappear. That's not what we intend, right?

**Review by johnwatkins0 — CHANGES_REQUESTED** [review]
Left a suggestion on this, @pierlon

File: assets/src/block-editor/helpers/index.js (hunk 1 above)
Comment (pierlon) [reply assets/src/block-editor/helpers/index.js]: Yea that's true, nice catch. I've addressed that in https://github.com/ampproject/amp-wp/pull/5574/commits/b3909733647e9e8a0096bf9e39816b81e94667f0.

**Review by johnwatkins0 — APPROVED** [review]

**Review by johnwatkins0 — APPROVED** [review]

**Review by westonruter — CHANGES_REQUESTED** [review]
As noted in https://github.com/ampproject/amp-wp/pull/5573#pullrequestreview-550612084, please add a deprecation notice that warns the feature will be removed.

File: assets/src/block-editor/helpers/index.js [hunk 2]
```diff
@@ -386,17 +386,32 @@ const AmpLayoutControl = ( props ) => {
...
-		/>
+		<>
+			<Notice
+				status="warning"
+				isDismissible={ false }
+			>
+				<span dangerouslySetInnerHTML={ {
+					__html: sprintf(
+						/* translators: placeholder is link to support forum. */
+						__( 'The AMP Layout setting is deprecated and is slated for removal. Please <a href="%s" target="_blank" rel="noreferrer">report</a> if you need it.', 'amp' ),
```
Comment (westonruter) [inline assets/src/block-editor/helpers/index.js]: This seems to be copy/pasta from #5573. Shouldn't it reference “AMP Noloading” and be put on `AmpNoloadingToggle` below instead?

**Review by westonruter — CHANGES_REQUESTED** [review]

File: assets/src/block-editor/helpers/index.js (hunk 2 above)
Comment (pierlon) [reply assets/src/block-editor/helpers/index.js]: Yes it should :facepalm:. Resolved in https://github.com/ampproject/amp-wp/pull/5574/commits/d8a4e0cc238821d790c6e5c2c2c292e9bfb8ca3b.

**Review by westonruter — APPROVED** [review]
For a newly-inserted image block, there is now no Noloading setting nor any Layout setting:

image

For a block that has those settings selected, the warnings appear:

image

**Pull request merged**